app/serializers/v2/namespace_serializer.rb

```diff
@@ -6,15 +6,17 @@ class NamespaceSerializer < ApplicationSerializer
     attribute :name
 
     link :self do
-      url_for(controller: 'v2/namespaces', action: 'show',
-              slug: object.to_param)
+      path = url_for(controller: 'v2/namespaces', action: 'show',
+                     slug: object.to_param, only_path: true)
+      [Settings.server_url, path].join('/')
```

Comment:
These will be simplified later by models knowing their own path; For the meantime, this seems to be the easiest solution, even if we're duplicating some code.